src/workerd/jsg/modules.h

```diff
@@ -409,25 +409,53 @@ class ModuleRegistryImpl final: public ModuleRegistry {
           if (type == Type::BUILTIN && entries.find(Key(path, Type::BUNDLE)) != kj::none) {
             continue;
           }
-
-          // The body of this callback is copied from `compileWasmGlobal` in src/workerd/server/workerd-api.c++.
-          addBuiltinModule(specifier, [specifier, module, this](Lock& lock) {
-            lock.setAllowEval(true);
-            KJ_DEFER(lock.setAllowEval(false));
-
-            // Allow Wasm compilation to spawn a background thread for tier-up, i.e. recompiling
-            // Wasm with optimizations in the background. Otherwise Wasm startup is way too slow.
-            // Until tier-up finishes, requests will be handled using Liftoff-generated code, which
-            // compiles fast but runs slower.
-            AllowV8BackgroundThreadsScope scope;
-            auto wasmModule = jsg::compileWasmModule(lock, module.getWasm().asBytes(), this->observer);
-            return jsg::ModuleRegistry::ModuleInfo(
-                  lock,
-                  specifier,
-                  kj::none,
-                  jsg::ModuleRegistry::WasmModuleInfo(lock, wasmModule));
-          }, type);
-          continue;
+          switch(module.which()) {
+          case Module::WASM:
+            // The body of this callback is copied from `compileWasmGlobal` in src/workerd/server/workerd-api.c++.
+            addBuiltinModule(specifier, [specifier, module, this](Lock& lock) {
+              lock.setAllowEval(true);
+              KJ_DEFER(lock.setAllowEval(false));
+
+              // Allow Wasm compilation to spawn a background thread for tier-up, i.e. recompiling
+              // Wasm with optimizations in the background. Otherwise Wasm startup is way too slow.
+              // Until tier-up finishes, requests will be handled using Liftoff-generated code, which
+              // compiles fast but runs slower.
+              AllowV8BackgroundThreadsScope scope;
+              auto wasmModule = jsg::compileWasmModule(lock, module.getWasm().asBytes(), this->observer);
+              return jsg::ModuleRegistry::ModuleInfo(
+                    lock,
+                    specifier,
+                    kj::none,
+                    jsg::ModuleRegistry::WasmModuleInfo(lock, wasmModule));
+            }, type);
+            continue;
+          case Module::DATA:
+            addBuiltinModule(specifier, [specifier, module](Lock& lock) {
+              auto value = kj::heapArray(module.getData().asBytes().asBytes());
+              v8::Local<v8::ArrayBuffer> data;
+              {
+                // Code duplicated from ArrayBufferWrapper::wrap in value.h because I
+                // couldn't figure out how to use ArrayBufferWrapper directly.
+                // TODO: Avoid code duplication
+                byte* begin = value.begin();
```

Comment:
Look at `jsg::Lock::wrapBytes(...)` ... it should do what you need here.

Reply:
Thanks!